src/test/java/org/jenkinsci/plugins/workflow/steps/EnvStepTest.java

```diff
@@ -158,6 +245,44 @@
         });
     }
 
+    @Test public void mapNumbersNested() {
```

Comment:
Some simpler variant of this may still be useful, to assert that map syntax allows non-string values.